components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -0,0 +1,1299 @@
+  // ------------------------------------------------------------------------
+  // ice nucleation output
+  naai_ = get_field_out("ni_activated").get_view<Real **>();
+
+  // droplet activation output
+  tendnd_ = get_field_out("nc_nuceat_tend").get_view<Real **>();
+
+  // interstitial and cloudborne aerosol tracers of interest: mass (q) and
+  // number (n) mixing ratios
+  for(int m = 0; m < mam_coupling::num_aero_modes(); ++m) {
+    // interstitial aerosol tracers of interest: number (n) mixing ratios
+    const char *int_nmr_field_name = mam_coupling::int_aero_nmr_field_name(m);
+    wet_aero_.int_aero_nmr[m] =
+        get_field_out(int_nmr_field_name).get_view<Real **>();
+    dry_aero_.int_aero_nmr[m] = buffer_.dry_int_aero_nmr[m];
+
+    // cloudborne aerosol tracers of interest: number (n) mixing ratios
+    const char *cld_nmr_field_name = mam_coupling::cld_aero_nmr_field_name(m);
+    wet_aero_.cld_aero_nmr[m] =
+        get_field_out(cld_nmr_field_name).get_view<Real **>();
+    dry_aero_.cld_aero_nmr[m] = buffer_.dry_cld_aero_nmr[m];
+
+    for(int a = 0; a < mam_coupling::num_aero_species(); ++a) {
+      // (interstitial) aerosol tracers of interest: mass (q) mixing ratios
+      const char *int_mmr_field_name =
+          mam_coupling::int_aero_mmr_field_name(m, a);
+
+      if(strlen(int_mmr_field_name) > 0) {
+        wet_aero_.int_aero_mmr[m][a] =
+            get_field_out(int_mmr_field_name).get_view<Real **>();
+        dry_aero_.int_aero_mmr[m][a] = buffer_.dry_int_aero_mmr[m][a];
+      }
+
+      // (cloudborne) aerosol tracers of interest: mass (q) mixing ratios
+      const char *cld_mmr_field_name =
+          mam_coupling::cld_aero_mmr_field_name(m, a);
+      if(strlen(cld_mmr_field_name) > 0) {
+        wet_aero_.cld_aero_mmr[m][a] =
+            get_field_out(cld_mmr_field_name).get_view<Real **>();
+        dry_aero_.cld_aero_mmr[m][a] = buffer_.dry_cld_aero_mmr[m][a];
+      }
+    }
+  }
+  for(int g = 0; g < mam_coupling::num_aero_gases(); ++g) {
+    const char *gas_mmr_field_name = mam_coupling::gas_mmr_field_name(g);
+    wet_aero_.gas_mmr[g] =
+        get_field_out(gas_mmr_field_name).get_view<Real **>();
+    dry_aero_.gas_mmr[g] = buffer_.dry_gas_mmr[g];
+  }
+
+  // hetrozenous freezing outputs
+  hetfrz_immersion_nucleation_tend_ =
+      get_field_out("hetfrz_immersion_nucleation_tend").get_view<Real **>();
+  hetfrz_contact_nucleation_tend_ =
+      get_field_out("hetfrz_contact_nucleation_tend").get_view<Real **>();
+  hetfrz_depostion_nucleation_tend_ =
+      get_field_out("hetfrz_depostion_nucleation_tend").get_view<Real **>();
+
+  //---------------------------------------------------------------------------------
+  // Allocate memory for the class members
+  // (Kokkos::resize only works on host to allocates memory)
+  //---------------------------------------------------------------------------------
+
+  Kokkos::resize(rho_, ncol_, nlev_);
```

Comment:
Does it make a difference if this view is allocated as  `rho_= view_2d("rho", ncol_, nlev_); `

Reply:
I don't think so. @overfelt  uses resize and I think both methods may be equivalent.

Reply:
Yes, both methods are the same since we know that rho_ is a view_2d.  In general Kokkos::resize would work if it was a view of ints or some other declaration that would need to be duplicated if resized without using Kokkos::resize.